core/types/cairo_tx.go

```diff
@@ -0,0 +1,181 @@
+	}
+	if len(b) > 32 {
+		return fmt.Errorf("wrong size for MaxFeePerGas: %d", len(b))
+	}
+	//tx.FeeCap = new(uint256.Int).SetBytes(b)
+	if tx.Gas, err = s.Uint(); err != nil {
+		return err
+	}
+	if b, err = s.Bytes(); err != nil {
+		return err
+	}
+	if len(b) > 0 && len(b) != 20 {
+		return fmt.Errorf("wrong size for To: %d", len(b))
+	}
+	if len(b) > 0 {
+		tx.To = &common.Address{}
+		copy((*tx.To)[:], b)
+	}
+	if b, err = s.Bytes(); err != nil {
+		return err
+	}
+	if len(b) > 32 {
+		return fmt.Errorf("wrong size for Value: %d", len(b))
+	}
+	tx.Value = new(uint256.Int).SetBytes(b)
+	if tx.Data, err = s.Bytes(); err != nil {
+		return err
+	}
+	// decode AccessList
+	//tx.AccessList = AccessList{}
+	//if err = decodeAccessList(&tx.AccessList, s); err != nil {
+	//	return err
+	//}
+	// decode V
+	if b, err = s.Bytes(); err != nil {
+		return err
+	}
+	if len(b) > 32 {
+		return fmt.Errorf("wrong size for V: %d", len(b))
+	}
+	tx.V.SetBytes(b)
+	if b, err = s.Bytes(); err != nil {
+		return err
+	}
+	if len(b) > 32 {
+		return fmt.Errorf("wrong size for R: %d", len(b))
+	}
+	tx.R.SetBytes(b)
+	if b, err = s.Bytes(); err != nil {
+		return err
+	}
+	if len(b) > 32 {
+		return fmt.Errorf("wrong size for S: %d", len(b))
+	}
+	tx.S.SetBytes(b)
+	return s.ListEnd()
+}
+
+func (tx CairoTransaction) GetChainID() *uint256.Int {
+	panic("implement me")
+}
+
+func (tx CairoTransaction) GetPrice() *uint256.Int {
+	panic("implement me")
+}
+
+func (tx CairoTransaction) GetTip() *uint256.Int {
+	panic("implement me")
+}
+
+func (tx CairoTransaction) GetEffectiveGasTip(baseFee *uint256.Int) *uint256.Int {
+	panic("implement me")
+}
+
+func (tx CairoTransaction) GetFeeCap() *uint256.Int {
+	panic("implement me")
+}
+
+func (tx CairoTransaction) Cost() *uint256.Int {
+	panic("implement me")
+}
+
+func (tx CairoTransaction) AsMessage(s Signer, baseFee *big.Int) (Message, error) {
+	panic("implement me")
+}
+
+func (tx CairoTransaction) WithSignature(signer Signer, sig []byte) (Transaction, error) {
+	panic("implement me")
+}
+
+func (tx CairoTransaction) FakeSign(address common.Address) (Transaction, error) {
+	panic("implement me")
+}
+
+func (tx CairoTransaction) Hash() common.Hash {
+	panic("implement me")
+}
+
+func (tx CairoTransaction) SigningHash(chainID *big.Int) common.Hash {
+	panic("implement me")
+}
+
+func (tx CairoTransaction) Size() common.StorageSize {
+	panic("implement me")
+}
+
+func (tx CairoTransaction) GetAccessList() AccessList {
+	panic("implement me")
+}
+
+func (tx CairoTransaction) Protected() bool {
+	panic("implement me")
+}
+
+func (tx CairoTransaction) RawSignatureValues() (*uint256.Int, *uint256.Int, *uint256.Int) {
+	panic("implement me")
+}
+
+func (tx CairoTransaction) MarshalBinary(w io.Writer) error {
+	panic("implement me")
+}
+
+func (tx CairoTransaction) Sender(signer Signer) (common.Address, error) {
+	panic("implement me")
+}
+
+func (tx CairoTransaction) GetSender() (common.Address, bool) {
+	panic("implement me")
```

Comment:
can we somehow reuse what the regulat tx uses? 
but that could be done as a follow-up PR

Reply:
Move implementation to https://github.com/ledgerwatch/erigon/blob/62b4294856de54348a28a795b07dc58b691f58fd/core/types/transaction.go#L108